src/parser_fill_cmd.c

```
@@ -0,0 +1,168 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parser_fill_cmd.c                                  :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/07/22 14:59:22 by sqiu              #+#    #+#             */
+/*   Updated: 2023/07/26 01:15:14 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file parser_fill_cmd.c
+ * @brief Functions to fill the cmd struct.
+ */
+
+#include "mod_parser.h"
+
+/**
+ * @brief Open file and save fd as fd_in.
+ * 
+ * @param lst 		List of token.
+ * @param new 		New cmd to be filled.
+ * @return t_err 	ERR_BAD_FD, SUCCESS
+ */
+t_err	ft_save_infile(t_tkn_list **lst, t_cmd *new)
+{
+	int			fd_in;
+	t_tkn_list	*tmp;
+
+	tmp = *lst;
+	tmp = tmp->next;
+	new->infile = 1;
+	if (new->fd_in > 0)
+		close(new->fd_in);
```

Technically we should also check the return of close. But I'm not quite sure what happens if it fails :D

---

It should return an error as well and lead to program abortion. Implemented it